src/anemoi/models/layers/attention.py

```diff
@@ -89,24 +139,220 @@ def forward(
         value = shard_heads(value, shapes=shapes, mgroup=model_comm_group)
         dropout_p = self.dropout_p if self.training else 0.0
 
-        if _FLASH_ATTENTION_AVAILABLE:
-            query, key, value = (
-                einops.rearrange(t, "batch heads grid vars -> batch grid heads vars") for t in (query, key, value)
+        out = self.attention(
+            query,
+            key,
+            value,
+            batch_size,
+            causal=False,
+            window_size=self.window_size,
+            dropout_p=dropout_p,
+            softcap=self.softcap,
+            alibi_slopes=self.alibi_slopes,
+        )
+
+        out = shard_sequence(out, shapes=shapes, mgroup=model_comm_group)
+        out = einops.rearrange(out, "batch heads grid vars -> (batch grid) (heads vars)")
+
+        out = self.projection(out)
+
+        return out
+
+
+class TorchAttentionWrapper(nn.Module):
+    """Wrapper for Pytorch dot product attention"""
+
+    def __init__(self):
+        super().__init__()
+
+        from torch.nn.functional import scaled_dot_product_attention
+
+        self.attention = scaled_dot_product_attention
+        self.mask = None
+        self.window_size = None
+
+    def update_mask(self, seq_len, window_size: int, device: str):
+        update_mask = (
+            self.mask is None or self.window_size != window_size or tuple(self.mask.shape) != (seq_len, seq_len)
```

Comment:
This could use a comment to explain the logic.

I find it a bit weird that we call `update_mask` and sometimes it doesn't update the mask without informing the user in any way.